### PR TITLE
Add option to disable root privileges nag

### PR DESCRIPTION
**Pull request by alexiri**
Recently, an "external access" warning was introduced, and @garfield69 implemented a button to dismiss the warning. As [pointed out here](https://github.com/Jackett/Jackett/issues/15066#issuecomment-1953129520), it is the user's responsibility to understand and accept the consequences of ignoring said warning.

In the same light, I would like to introduce a way to dismiss the warning about running with root privileges. The user has been warned of the risks and they should be able to dismiss the warning under their responsibility.

### CODE REVIEW

File: src/Jackett.Server/Services/ServerService.cs [hunk 1]
```diff
@@ -274,7 +274,7 @@ public void Initialize()
 
             try
             {
-                if (Environment.UserName == "root")
+                if (Environment.UserName == "root" && String.IsNullOrEmpty(Environment.GetEnvironmentVariable("DisableRootWarning")))
```
Comment (garfield69) [inline src/Jackett.Server/Services/ServerService.cs:277]: should we also document this someplace, in the Readme or on the wiki?

**Comment by garfield69** [comment]
I have added a  wiki entry for this https://github.com/Jackett/Jackett/wiki/Troubleshooting/#jackett-is-running-with-root-privileges-you-should-run-jackett-as-an-unprivileged-user
Please read and confirm that it is factual and correct and if good let me know and I will merge this PR.

**Comment by alexiri** [comment]
The wiki entry looks good! Thanks!

**Comment by alexiri** [comment]
Nobody is encouraging it. There's also no need to be paternalistic: the warning is there, the user has to go out of their way to suppress it, let them do it if they want to. Why die on this hill?

**Comment by alexiri** [comment]
> Also which is incorrect, it allows 0 or false to be treated as true.

Fair point, should be fixed now.

**Comment by garfield69** [comment]
Thank you.

**Comment by garfield69** [comment]
v0.21.2441

**Pull request merged**